### ISSUE TITLE
.ply File compression

### ISSUE THREAD

**Issue by imapoorvaholla** [Esri/lepcc#11]
Hello Team,
Please may I know is it possible to compress .ply(point cloud data) file by using LEPCC compression method. If yes please may I know the process.

Looking forward to hearing from you.
Thank You
Reagards,
Apoorva

**Comment by tmaurer3** [comment]
@imapoorvaholla  It depends. Lepcc compresses 3D points (xyz coo, RGB color, intensity if there) but not meshes, surface normals etc. If you want to store the mesh, the answer is no. If you are fine with the 3D points + RGB colors of the points, then you can. But it's not a button press. Lepcc assumes a large point cloud with billions of points gets tiled or boxed up into many 3D boxes (by some higher level code). Then each such box gets encoded into one compressed Lepcc blob. If your dataset is not too big, you can try to compress into one single Lepcc blob. 

The main purpose of this repository is to allow interested parties to write their own decoders for 3D point clouds encoded in Esri's slpk format. However, the Lepcc API also allows to encode. But you would need to convert the 3D points from your .ply file into simple arrays first. Then you can Lepcc encode it. There is sample code for this here:

https://github.com/Esri/lepcc/blob/master/src/Test_C_Api.cpp#L277

In a nutshell, you can provide the data as a sequence of pairs of arrays, always 1st array with the xyz coo as double, 2nd array with the RGB colors as byte. Then you call the API encode functions with these arrays as input. Check the output size, test the compression ratio, write out the compressed binary blobs into an outgoing file. You can and should add your own header if you do that.
